File: daily/2026/2144_Minimum_Cost_of_Buying_Candies_With_Discount.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int minimumCost(vector<int>& cost) {
        int n = cost.size();
        sort(cost.begin(), cost.end());
        int ans = 0;
        int buy = 0;
        for (int i = n - 1; i >= 0; i--){
            if (buy < 2){
                ans += cost[i];
                buy++;
            }
            else{
                buy = 0;
            }
        }
        return ans;
    }
};

int main(){
    vector<int> cost = {1,2,3};
    // vector<int> cost = {6,5,7,9,2,2};
    // vector<int> cost = {5,5};
    Solution S;
    cout << S.minimumCost(cost) << endl;
    return 0;
}

File: daily_python/2144_Minimum_Cost_of_Buying_Candies_With_Discount.py
import sys
import os
from typing import List

class Solution:
    def minimumCost(self, cost: List[int]) -> int:
        cost.sort(key = lambda x : -x)
        ans = sum(cost)
        n = len(cost)
        for i in range(2, n, 3):
            ans -= cost[i]
        return ans
    
def readData(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            cost = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(cost)
    return dataset

if __name__ == '__main__':
    if (len(sys.argv) == 1):
        path = os.path.basename(__file__)
        filename = os.path.splitext(path)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readData(filename)
    results = []
    solution = Solution()
    for cost in dataset:
        results.append(solution.minimumCost(cost))
    for index, result in enumerate(results):
        print(f'dataset {index+1} : {result}')
